core/services/workflows/syncer/v2/workflow_registry.go

```diff
@@ -189,20 +193,32 @@ func (w *workflowRegistry) Start(_ context.Context) error {
 			reader, err := w.newWorkflowRegistryContractReader(ctx)
 			if err != nil {
 				w.lggr.Criticalf("contract reader unavailable : %s", err)
+				w.hooks.OnStartFailure(err)
+				cancel()
 				return
 			}
 
 			w.contractReader = reader
+			close(initDoneCh)
```

Comment:
only close `initDoneCh` if the `reader` is initialized, otherwise the context is canceled and the waiting routines will stop.